Controllers/DatasetController.py

```diff
@@ -138,16 +135,14 @@ def search(searchQuery):
                             DatasetService.createDatasetInfoObject(correspondingDataset))
                     except:
                         pass
-
                 matchedDatasets = Dataset.objects.search_text(
                     searchQuery).order_by('$text_score')
                 for dataset in matchedDatasets:
```

Comment:
Don't repeat this code block (for dataset in matchedDatasets). Determine how to write this code block only once for both cases (user only and all)